tests: flaky ssh tests

https://travis-ci.com/github/iterative/dvc/jobs/340412779

```sh
tests/unit/remote/ssh/test_connection.py::test_hardlink 
[gw1] [ 80%] FAILED tests/unit/remote/ssh/test_connection.py::test_hardlink 
tests/unit/remote/ssh/test_connection.py::test_copy 
[gw1] [ 80%] FAILED tests/unit/remote/ssh/test_connection.py::test_copy 
```

I thought this was only failing on my machine, so I was ignoring it. But, looks like it occasionally happens on the CI. On my machine though this always appears.

Also, an important thing to note here is that, locally on my machine. it only fails when I'm running all of the tests, but when I rerun failed tests, it passes.

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -34,7 +34,7 @@ user = "user"
 key_path = os.path.join(here, f"{user}.key")
 
 
-@pytest.fixture
+@pytest.fixture(scope="session")
 def ssh_server():
     users = {user: key_path}
     with mockssh.Server(users) as s:

